core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1334,18 +1334,19 @@ public void install() {
          *      the plugin will only take effect after the reboot.
          *      See {@link UpdateCenter#isRestartRequiredForCompletion()}
          * @param correlationId A correlation ID to be set on the job.
+         * @param batch if defined, a list of plugins to add to, which will be started later
          */
         @Restricted(NoExternalUse.class)
-        public Future<UpdateCenterJob> deploy(boolean dynamicLoad, @CheckForNull UUID correlationId) {
+        public Future<UpdateCenterJob> deploy(boolean dynamicLoad, @CheckForNull UUID correlationId, @CheckForNull List<PluginWrapper> batch) {
```

Comment:
have you checked API usages there. It was not Restricted before https://github.com/jenkinsci/jenkins/pull/2269 by @daniel-beck

Reply:
From plugins built against Jenkins 1.x? That is pretty old. This is also not a sort of API generally expected to be used by plugins. AFAIK we have no tool to look for such method calls structurally (the method name is too generic to search for).

Reply:
```console
user@jenkinsci$ find *-plugin/src -type f -print0 | xargs -0 egrep -l 'UpdateSite|UpdateCenter' | xargs fgrep deploy
blackduck-installer-plugin/src/main/java/com/blackducksoftware/integration/jenkins/BDInstaller.java:                deployPlugin(vulnerabilityReportPlugin);
blackduck-installer-plugin/src/main/java/com/blackducksoftware/integration/jenkins/BDInstaller.java:    private static void deployPlugin(final UpdateSite.Plugin plugin)
blackduck-installer-plugin/src/main/java/com/blackducksoftware/integration/jenkins/BDInstaller.java:        plugin.deploy(true);
cloudbees-enterprise-plugins-plugin/src/main/java/com/cloudbees/jenkins/plugins/enterpriseplugins/PluginImpl.java:            require("cloudbees-aws-deployer","1.13"),
cloudbees-enterprise-plugins-plugin/src/main/java/com/cloudbees/jenkins/plugins/enterpriseplugins/PluginImpl.java:                                p.deploy().get();
cloudbees-enterprise-plugins-plugin/src/main/java/com/cloudbees/jenkins/plugins/enterpriseplugins/PluginImpl.java:                            p.deploy().get();
recipe-plugin/src/main/java/org/jenkinsci/plugins/recipe/ingredients/PluginIngredient.java:                        installSource.deploy(true).get();
recipe-plugin/src/main/java/org/jenkinsci/plugins/recipe/ingredients/PluginIngredient.java:                            installSource.deploy(true).get();
system-config-dsl-plugin/src/main/java/org/jenkinsci/plugins/systemconfigdsl/Root.java:            jenkins.getUpdateCenter().getPlugin(name).deploy(true).get();
thin-backup-plugin/src/main/java/org/jvnet/hudson/plugins/thinbackup/restore/HudsonRestore.java:              return plugin.deploy();
```

does not indicate any such calls. This sort of search is of course imprecise since an `UpdateSite` object might have been obtained via method chaining without the type being imported, but the only way I can see this happening is via getters from an `UpdateCenter`, which is turn can only be obtained via a `getUpdateCenter` method that would also match.